src/pages/LoginPage.tsx

```diff
@@ -16,31 +14,50 @@ import { useNavigate } from "react-router";
 import { SubmitHandler, useForm } from "react-hook-form";
 import { useSelector } from "react-redux";
 import { CircularProgress } from "@mui/material";
-
 import { RootState } from "../redux/store";
 import { UserLoginData } from "../redux/auth/auth.type";
 import { userLogin } from "../redux/auth/auth.actions";
+import { useEffect, useState } from "react";
+import { theme } from "../styles/theme";
 
-const defaultTheme = createTheme();
+const defaultTheme = createTheme(theme);
 
 export const Login = (): JSX.Element => {
   const dispatch = useAppDispatch();
   const navigate = useNavigate();
   const auth = useSelector((state: RootState) => state.auth);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<UserLoginData>();
 
   const onSubmit: SubmitHandler<UserLoginData> = async (data) => {
-    await dispatch(userLogin(data));
-    navigate("/");
+    try {
+      await dispatch(userLogin(data));
+      console.log(auth);
+    } catch (error) {
+      if (error instanceof Error) {
+        setErrorMessage(error.message);
+      } else {
+        setErrorMessage("Invalid username or password. Please try again");
+      }
+      setTimeout(() => window.location.reload(), 3000);
```

Comment:
Hmm why are you purposely slowing down the login?

Reply:
It was meant to reload the page after displaying the message for 3 seconds so that the user could try again, but this has become redundant, as they can just immediately input new credentials